plugins/azure-devops/src/alpha/plugin.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  createApiExtension,
+  createApiFactory,
+  createPageExtension,
+  createPlugin,
+  discoveryApiRef,
+  identityApiRef,
+} from '@backstage/frontend-plugin-api';
+import { azureDevOpsApiRef, AzureDevOpsClient } from '../api';
+import {
+  compatWrapper,
+  convertLegacyRouteRef,
+} from '@backstage/core-compat-api';
+import {
+  createEntityCardExtension,
+  createEntityContentExtension,
+} from '@backstage/plugin-catalog-react/alpha';
+import { azurePullRequestDashboardRouteRef } from '../routes';
+
+/** @alpha */
+export const azureDevOpsApi = createApiExtension({
+  factory: createApiFactory({
+    api: azureDevOpsApiRef,
+    deps: { discoveryApi: discoveryApiRef, identityApi: identityApiRef },
+    factory: ({ discoveryApi, identityApi }) =>
+      new AzureDevOpsClient({ discoveryApi, identityApi }),
+  }),
+});
+
+/** @alpha */
+export const azureDevOpsPullRequestPage = createPageExtension({
+  defaultPath: '/azure-pull-requests',
+  routeRef: convertLegacyRouteRef(azurePullRequestDashboardRouteRef),
+  loader: () =>
+    import('../components/PullRequestsPage').then(m =>
+      compatWrapper(<m.PullRequestsPage />),
```

Comment:
I see this component expects a few props. Would you like to make them configurable via `app-config.yaml` file? If that sounds like a good idea, you can define a config schema for this page.
Here is an example:
https://github.com/awanlin/backstage/blob/cfe2e426cfbbc73e55c9e94033536d8679f54388/plugins/search/src/alpha.tsx#L103-L108

Reply:
Going to answer all the props related comments here. So, when I approached this I assumed that I needed to do this in config but I went and created a PR to do it in the way I know - https://github.com/backstage/backstage/pull/22046 - but from the sounds of it that's not the intended pattern for the New Frontend System. I'll work on updating this to use the new pattern 👍

Reply:
@camilaibs do you have an example of using config like this using `createEntityContentExtension`? Also with `createEntityCardExtension`? I seem to hit a type error when I add `configSchema` to these :(

Reply:
Hello @awanlin, as I mentioned here, `createEntityContentExtension` does not accept a `configSchema`, you would have to update its options for it to start receiving the schema:
https://github.com/backstage/backstage/blob/62c54c17815759d1310f6dd133925fcd51af1e27/plugins/catalog-react/src/alpha.tsx#L108-L121

Reply:
I gave that a try by adding `configSchema` as optional and then doing a conditional check but got really confused by `<TInputs>` and `<TOutputs>`. This more due to a weakness on my part then the code being unclear.